### ISSUE TITLE
False negative: 'group-labelledby' check in 'checkboxgroup' rule does not fail corner cases

### ISSUE THREAD

**Issue by iamrafan** [dequelabs/axe-core#1111]
Screen reader experience for the following corner cases is not as expected, hence the ```checkboxgroup``` rule should fail

- Case-1: Ungrouped checkboxes with same name and each having aria-labelledby pointing to a common label

- Case-2: Ungrouped checkboxes with explicit [label] and each having aria-labelledby pointing one of the checkbox labels

- Case-3: Ungrouped checkboxes with implicit [label] and each having aria-labelledby pointing one of the checkbox labels

**Link to live demo:** https://codepen.io/iamrafan/pen/MqjwOz

**AT test results:**
![image](https://user-images.githubusercontent.com/22068434/44758523-bd9bb400-aae9-11e8-90df-08302a7be46e.png)

<pre><code>axe-core version: 3.0.3
axe-Coconut version: 3.4.0-alpha.2</code></pre>

**Comment by WilcoFiers** [comment]
Exploring this a little bit. I think the problem with group-labelledby is that it doesn't check if there is also a unique label. The following code shouldn't pass:

```html
<input type="radio" name="foo" aria-labelledby="bar" />
<input type="radio" name="foo" aria-labelledby="bar" />
<p id="bar">Label</p>
```

As for the odd Firefox thingy, that's a known issue. Our accessible name computation algorithm takes that into consideration.
